src/System.Management.Automation/engine/CoreAdapter.cs

```diff
@@ -4252,7 +4252,7 @@ protected override T GetMember<T>(object obj, string memberName)
             // are ignored.
             if (typeof(T) == typeof(PSMemberInfo))
             {
-                T returnValue = PSObject.dotNetInstanceAdapter.GetDotNetMethod<T>(obj, memberName);
+                T returnValue = base.GetDotNetMethod<T>(obj, memberName);
```

Comment:
Do you have an example that demonstrates this is a bug?

It might be a change in behavior, e.g. if the instance was a "property only" adapter, `GetMember` might no longer return any methods whereas it would have before.

Reply:
> if the instance was a "property only" adapter

This type is `BaseDotNetAdapterForAdaptedObjects` and it derives from `DotNetAdapter` directly. An instance of it won't be a "property only" adapter. It serves as the fall-back DotNet adapter for custom adapters.
I may miss an aspect of your comment, could you please explain your concern a bit more?

Reply:
And performance shouldn't be a problem because the cache tables are static to the process, so every instance of `DotNetAdapter` can benefit from them.

Reply:
I see, I thought this might be used by `ThirdPartyAdapter` or other `PropertyOnlyAdapter` implementations, but it's the fallback adapter.

After reviewing `BaseDotNetAdapterForAdaptedObjects` and `PropertyOnlyAdapter` and related methods, I don't understand why it exists at all.

Reply:
I have made the similar change to `PropertyOnlyAdapter`.